macros/get_metric_sql.sql

```diff
@@ -48,33 +48,23 @@ LETS SET SOME VARIABLES!
 a custom calendar #}
 {%- set calendar_tbl = ref(var('dbt_metrics_calendar_model', "dbt_metrics_default_calendar")) %}
 
-{# Here we are creating the dimension list which has the list of all dimensions that 
-are a part of the metric. This has additional logic for multiple metrics #}
-    {# TODO #48 get valid common dimension name to clarify what is #}
-{%- set dimension_list = [] -%}
-{% for metric in metric_list %}
-    {# get all dimensions in metric  #}
-    {%- set metric_dimensions= metrics.get_valid_dimension_list(metric) -%}
-    {# This line removes any of the dimensions for loop metric that don't exist in the dimension list already
-    This creates a smaller list that only contains the subset #}
-
-    {# TODO #50 build common list across all metrics and then find ones that appear X (metric count) times  #}
-    {%- set new_dimensions = ( metric_dimensions | reject('in',dimension_list) | list) -%}
-    {% for dim in new_dimensions %}
-        {%- do dimension_list.append(dim) -%}
-    {% endfor %}
-{%endfor%}
+{# Here we are creating a list of all valid dimensions, as well as providing compilation
```

Comment:
I think this macro is doing too much. I would recommend having one macro that _gets_ the list of all dimensions, and another that checks they are legal. 

From my reading of this, you're passing in the `dimensions`, and then getting back a list of valid dimensions (`common_valid_dimension_list`), but if any `dimensions` aren't valid then you throw a compiler error. So am I correct to assume that the available outcomes are 
- you get `dimensions` back as-provided, and give it a new name
- you get an exception
?

If so, you could do something like this:
```
set all_dimensions = get_all_dimensions(whatever_args_you_need)
validate_dimension_universality(all_dimensions, any_other_args_you_need)
```